src/components/Dropzone/Dropzone.vue

```diff
@@ -0,0 +1,345 @@
+
+export default {
+  name: 'Dropzone',
+  components: { LobLink, ProgressBar, Upload },
+  props: {
+    textContent: { type: Object, required: true,
+      validator: function (value) {
+        const requiredKeys = ['dragAndDropHere', 'yourFile', 'or', 'uploadFileButtonText',
+          'acceptedFormatIs', 'acceptedFormatsAre', 'maxFileSizeIs', 'downloadSampleFile',
+          'uploading', 'mightTakeAMinute', 'looksGreat', 'removeFileButtonText',
+          'couldNotUpload', 'canOnlySelectOneFile', 'fileIsTooLarge', 'fileTypeNotValid',
+          'defaultErrorText', 'errorMessage', 'successMessage'];
+        for (const key of requiredKeys) {
+          return (Object.keys(value).includes(key));
+        }
+      }
+    },
+    uploadId: { type: String, required: true },
+    acceptType: { type: String, default: '/*' },
+    maxSizeInBytes: { type: [Number, null], default: null },
+    showTypeAndMaxSize: { type: Boolean, default: true },
+    sampleLinkUrl: { type: String, default: null },
+    progress: { type: [Number, null], default: null },
+    fileOb: {  type: Object, default: null },
+    status: { type: [String, null], default: null,
+      validator: function (value) {
+        return [null, 'error', 'success'].includes(value);
+      }
+    }
+  },
+  emits: ['select', 'remove'],
+  data: function () {
+    return {
+      formatBytes,
+      selectedFile: this.fileOb,
+      currentStep: this.status !== null ? this.status :  'default',
+      dragOverlay: false,
+      multipleFilesError: false,
+      fileSizeError: false,
+      fileTypeError: false
+    };
+  },
+  computed: {
+    defaultStep () {
+      return this.currentStep === 'default';
+    },
+    errorStep () {
+      return this.currentStep === 'error';
+    },
+    uploadingStep () {
+      return this.currentStep === 'uploading';
+    },
+    successStep () {
+      return this.currentStep === 'success';
+    },
+    fileHasBeenSelected () {
+      return this.successStep && this.selectedFile;
+    },
+    selectedFileType () {
+      return this.selectedFile.type ? this.selectedFile.type.split('/')[1].toUpperCase() : 'File';
+    },
+    selectedFileName () {
+      return this.selectedFile.name ? this.selectedFile.name : 'File';
+    },
+    fileTypesArray () {
+      const array = this.acceptType?.split(',') || [];
+      return array.map((type) => type.trim());
+    },
+    fileTypesString () {
+      return this.fileTypesArray.map((t) => t.replace(/[^a-zA-Z]/g, '')).join(', ').toUpperCase();
+    },
+    isDroppingDisabled () {
+      return this.fileHasBeenSelected || this.uploadingStep;
+    },
+    displayText () {
+      if (this.errorStep) {
+        return this.textContent.couldNotUpload;
+      }
+      if (this.successStep) {
+        return this.textContent.looksGreat;
+      }
+      if (this.uploadingStep) {
+        return `${this.textContent.uploading} ${this.selectedFileName}...`;
+      } else {
+        return '';
+      }
```

Comment:
No need to else here, you can just move the `return '';` to the last line after the last `if` block.  The `else` is implicit because if it had been any of the other conditions the function would have returned before then.